python_scripts/metrics.py

```diff
@@ -355,13 +366,13 @@
 # plot_roc_curve(classifier, X_test, y_test, pos_label='donated')
 
 # %% [markdown]
-# This curve is built with the same principle than with the precision-recall
+# This curve is built using the same principle as the precision-recall
 # curve: we vary the probability threshold to compute "hard" prediction and
-# compute the metrics. As with the precision-recall curve as well, we can
+# compute the metrics. As with the precision-recall curve, we can
 # compute the area under the ROC (ROC-AUC) to characterize the performance of
-# our classifier. However, this is important to observer that the lower bound
-# of the ROC-AUC is 0.5. Indeed, we represented the performance of a dummy
-# classifier (i.e. green dashed line) to show that the worse performance
+# our classifier. However, it is important to observer that the lower bound
+# of the ROC-AUC is 0.5. Indeed, we show the performance of a dummy
```

Comment:
Maybe we could expand. I think the ROC-AUC value can be lower than 0.5 but 0.5 represents random chance level. And < 0.5 means the classifier is better at incorrectly classifying samples than chance and you should revert your predictions.

Reply:
Also we could give more intuition on ROC-AUC:
- When sensitivity = 1-specificity - it means our prediction is equally likely to be a true positive or false positive. Thus it represents chance.
- Ideal curve - as our chance of true positive increases we want our false positives to stay low as possible - we want the line to be as close y axis as possible.